### PR TITLE
Remove Pipeline and use ephemeral Pipelines in JES. Closes #856

### LINKED ISSUE #856: Switch JES to use ephemeral pipelines
From Dion, regarding how we construct a Pipeline for every Run:

Could you convert that into the ephemeral Run version?   RunPipelineRequest.ephemeral_pipeline instead of pipeline_id. Will save you one RPC call too.

### PR DESCRIPTION
**Pull request by geoffjentry**

### CODE REVIEW

**Comment by geoffjentry** [comment]
@mcovarr @Horneth - the code is slightly different than the hotfix to account for slightly different realities (plus a moving-stuff-around refactor) but more or less the same.

**Comment by Horneth** [comment]
👍

File: supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/RunStatus.scala [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package cromwell.backend.impl.jes
+
+sealed trait RunStatus {
+  import RunStatus._
+
+  // Could be defined as false for Initializing and true otherwise, but this is more defensive.
+  def isRunningOrComplete = this match {
+    case Running | _: TerminalRunStatus => true
```
Comment (mcovarr) [inline supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/RunStatus.scala]: that `_:` looks suspiciously superfluous

**Comment by mcovarr** [comment]
:+1:

File: supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/RunStatus.scala (hunk 1 above)
Comment (geoffjentry) [inline supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/RunStatus.scala]: Indeed. Someone named @mcovarr did it, we should ask him 😝

Comment (geoffjentry) [inline supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/RunStatus.scala]: Appears to simply look suspicious. The original culprit is vindicated. :shipit:

**Pull request merged**